2019/python/rules_engine.py

```diff
@@ -2368,56 +2368,55 @@ def v695(self):
-		and cannot be left blank.
-		Automated Underwriting System Result: 2; Automated Underwriting System Result: 3;
-		Automated Underwriting System Result: 4; and Automated Underwriting System Result: 5
-		must equal 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, or be left blank.
-
-		Impact of S2155: Update to: 
-		2) Automated Underwriting System Result: 1 must equal 1111, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, or 17, 
-		and cannot be left blank. 
-		Automated Underwriting System Result: 2; Automated Underwriting System Result: 3; Automated Underwriting System Result: 4; 
-		and Automated Underwriting System Result: 5 must equal 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, or be left blank. """
+		"""
+		2) Automated Underwriting System Result: 1 must
+		equal 1111, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14,
+		15, 16, 17, 18, 19, 20, 21, 22, 23, or 24 and cannot
+		be left blank. Automated Underwriting System
+		Result: 2; Automated Underwriting System Result: 3;
+		Automated Underwriting System Result: 4; and
+		Automated Underwriting System Result: 5 must
+		equal 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15,
+		16, 17, 18, 19, 20, 21, 22, 23, 24 or be left blank.
+		"""
 		field = "AUS 1-5 Result"
 		edit_name = "v696_2"
-		aus_1_results = ["1111", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17"]
-		aus_n_results = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", ""]
+		aus_1_results = ["1111", "1", "2", "3", "4", "5", "6", "7", 
+		"8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20",
+		"21", "22", "23","24"]
+		aus_n_results = ["1", "2", "3", "4", "5", "6", "7", 
+		"8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20",
+		"21", "22", "23","24", ""]
 		fail_df = self.lar_df[~(self.lar_df.aus_result_1.isin(aus_1_results))|(~self.lar_df.aus_result_2.isin(aus_n_results))|(~self.lar_df.aus_result_3.isin(aus_n_results))
 		|(~self.lar_df.aus_result_4.isin(aus_n_results))|(~self.lar_df.aus_result_5.isin(aus_n_results))]
 		self.results_wrapper(edit_name=edit_name, field_name=field, fail_df=fail_df)
 
 	def v696_3(self):
 		"""An invalid Automated Underwriting System data field was reported.
-		3) The number of reported Automated Underwriting Systems must equal the number of reported Automated Underwriting System Results."""
+		3) The number of reported Automated Underwriting Systems must 
+		equal the number of reported Automated Underwriting System Results."""
 		field = "AUS Systems and Results"
 		edit_name = "v696_3"
 		fields_1 = ["aus_1", "aus_2", "aus_3", "aus_4", "aus_5"]
 		fields_2 = ["aus_result_1", "aus_result_2", "aus_result_3", "aus_result_4", "aus_result_5"]
-		vals_1 = ("1", "2", "3", "4", "5")
-		vals_2 = ("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16")
+		vals_1 = ("1", "2", "3", "4", "5", "6", "")
```

Comment:
Is this passing blank as a valid value to AUS 1 in some cases?

Reply:
Last year's, `vals_1` values had the intersection of codes between `aus_1` and `aus_2`-`aus_5`. I was thinking of ways to include the union of codes (all possible ones for `aus_1` and `aus_2`-`aus_5`). I can revert back to only including the codes that can be in both `vals_1` and `vals_2`

Reply:
Alternatively you could create new vals lists that are more specific. I think that will give you better flexibility and also include all the valid values.

Maybe something like:
vals_aus_1 = [vals]
vals_aus_2_5 = [vals + blank]

This may require splitting out the fields_1 and fields_2 and handling them differently.

Reply:
Yes, I agree.

Reply:
Hi David, the problem here seems to be having blanks counted in `v696_3`. It causes the function to interpret a blank as though it were a reported value, which would cause an infinite loop.